daemon/state.go

```diff
@@ -216,6 +216,8 @@ func (d *Daemon) regenerateRestoredEndpoints(state *endpointRestoreState) {
 				return
 			}
 
+			ep.CreateBPFProgramChannel()
```

Comment:
The channel is leaked when the endpoint generation fails and the endpoint is never built again.

Reply:
By “leaked” do you mean that it persists even if the endpoint is deleted ? Does GC not take effect in that case? If the endpoint is never rebuilt , then its BPF program hasn’t been compiled and the channel is still open.

Reply:
Added code which closes the channel on endpoint deletion to ensure that the channel is GC'd when the endpoint is removed and that no one is blocking on it. Also removed the exported function to create the channel, and instead set its value internally within the Endpoint package. Let me know if this is what you had in mind.